File: FinTrack.API/Models/Loan.cs
// Konum: FinTrack.API/Models/Loan.cs
using System;

namespace FinTrack.API.Models
{
    public class Loan
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public int TargetAccountId { get; set; } // Kredi tutarının aktarılacağı hesap
        public string LoanType { get; set; } // "İhtiyaç", "Taşıt", "Konut"
        public decimal PrincipalAmount { get; set; } // Çekilen anapara
        public decimal InterestRate { get; set; } // Yıllık faiz oranı (örn: 0.60 for 60%)
        public int TermInMonths { get; set; } // Vade (ay olarak)
        public decimal MonthlyPayment { get; set; } // Aylık taksit tutarı
        public decimal TotalRepayment { get; set; } // Toplam geri ödeme tutarı
        public DateTime StartDate { get; set; }
        public bool IsActive { get; set; } = true;

        public virtual User User { get; set; }
        public virtual Account TargetAccount { get; set; }
    }
}

File: fintrack-client/src/types/loan.ts
// Konum: fintrack-client/src/types/loan.ts
export interface Loan {
    id: number;
    targetAccountName: string;
    loanType: string;
    principalAmount: number;
    interestRate: number; // Yüzde olarak gelir
    termInMonths: number;
    monthlyPayment: number;
    totalRepayment: number;
    startDate: string; // ISO Date String
    isActive: boolean;
}

export interface CreateLoanRequest {
    targetAccountId: number;
    loanType: string;
    amount: number;
    termInMonths: number;
}

// Hesaplama için yeni türler
export interface LoanCalculationRequest {
    amount: number;
    termInMonths: number;
    loanType: string;
}

export interface LoanCalculationResponse {
    monthlyPayment: number;
    totalRepayment: number;
    annualInterestRate: number;
}